src/module/MomentumDiffusion.cpp

```diff
@@ -0,0 +1,205 @@
+// TO DO 12/12/19 LM
+// 1.   Look into the limitNextStep function:
+//      It is quite complicated calculation and it is not clear if it will help
+//      Testing for this is required.
+// 2.a) Write a test suite for the unit tests
+// 	 b)	Write physics based tests
+// 3. Provide simple examples
+
+
+#include "crpropa/module/MomentumDiffusion.h"
+
+using namespace crpropa;
+
+ConstantMomentumDiffusion::ConstantMomentumDiffusion(double Dpp) {
+	setDpp(Dpp);
+}
+
+void ConstantMomentumDiffusion::process(Candidate *c) const {
+	double rig = c->current.getRigidity();
+	if (std::isinf(rig)) {
+		return; // Only charged particles
+	}
+	
+	double p = c->current.getEnergy()/c_light; // Note we use E=p/c (relativistic limit)
+	double dt = c->getCurrentStep() / c_light;
+	
+	std::cout <<dt<<"\n"; // Has to be deleted
+	double eta =  Random::instance().randNorm();
+	double domega = eta * sqrt(dt);
+	
+	double AScal = calculateAScalar(p);
+	double BScal = calculateBScalar();
+
+	double dp = AScal * dt + BScal * domega;
+	std::cout <<dp<<"\n"; //Has to be deleted
+	c->current.setEnergy((p + dp)*c_light);
+	
+	//Fast, but a little bit inconsitent, since the equation limit=\Delta p(step) / p
```

Comment:
Would it make sense to allow both modes here and add an option? Have you studied the difference?